contrib/inventory/ec2.py

```diff
@@ -1144,6 +1145,10 @@ def get_host_info_dict_from_instance(self, instance):
                     group_names.append(group.name)
                 instance_vars["ec2_security_group_ids"] = ','.join([str(i) for i in group_ids])
                 instance_vars["ec2_security_group_names"] = ','.join([str(i) for i in group_names])
+            elif key == 'ec2_block_device_mapping':
+                instance_vars["ec2_block_devices"] = {}
+                for k, v in value.iteritems():
+                    instance_vars["ec2_block_devices"][ ( k.split( '/' ) )[ -1 ] ] = v.volume_id
```

Comment:
Probably won't make any difference here but there's a stdlib function to split filename from directory name:
``` python
import os.path
os.path.basename(k)
```

or you can make this more targeted by doing it like this:

``` python
instance_vars["ec2_block_devices"][ k.rsplit( '/' )[ -1 ] ] = v.volume_id